src/components/panel/Panel.js

```diff
@@ -48,6 +48,7 @@ export default class Panel extends React.Component {
     open: bool,
     modal: bool,
     classes: object,
+    closeAction: node,
```

Comment:
Please remove this prop

Reply:
I did not see this! What is used for? The name makes sense for a function but for a node? I am curious

Reply:
@tureey I will try to explain :)
If we want to add a close button to the panel - 
```
<Panel
    trigger={<Button text="Open Modal" />}
     closable={true}
     modal={true}
     open={this.state.open}>
        <Panel.Header>Hello World!</Panel.Header>
        <Panel.Content>
            <p>To close the modal, use any of the buttons/actions</p>
         </Panel.Content>
         <Panel.Footer>
            <Button text="Deny" onClick={this.closeModalHandler} />
            <Button text="Accept" onClick={this.closeModalHandler} />
          </Panel.Footer>
</Panel>
```

Let's concentrate on `Panel.Footer` component.
It has two button components which handle the close action. The purpose of these buttons is to only close the panel and nothing else. For this we need to maintain the state of the panel in our component.

Consider a situation when we have multiple panels in the same component. We will have to change our component to stateful component (`class Comp extends React.Component{}`) and manage all the states in our Comp component. I feel this is an overhead to manage the state of Panel in the component, if we do not intend to do any action on the close of the panel.

I can change the name to something else.

Let me know if you have any questions.

Reply:
I think we mixed the idea of a Modal with a Panel. **For me, these are 2 different concepts.** A modal is something that can be **shown** or **hidden** and Panel is always visible.

Modal should be no more than one. Why? Because it's a portal. It's no more than a `<div>` that is used as a widget to inject HTML on demand. *In my opinion, we should have a HOC with functions that opens the portal given a title, a content and actions. and this hoc should be able to render the content provided to the Portal.*

To be pragmatic, I would talk about this once we have the problem you said and release with the changes that @crysfel was saying. We should **remember that iteration is the key**.

Reply:
> To be pragmatic, I would talk about this once we have the problem you said and release with the changes that @crysfel was saying. We should remember that iteration is the key.

👌